streaming/src/channel/channel.cc

```diff
@@ -0,0 +1,343 @@
+  MockQueue &mock_queue = MockQueue::GetMockQueue();
+  mock_queue.message_buffer[channel_info_.channel_id] =
+      std::make_shared<RingBufferImplThreadSafe<MockQueueItem>>(10000);
+  mock_queue.consumed_buffer[channel_info_.channel_id] =
+      std::make_shared<RingBufferImplThreadSafe<MockQueueItem>>(10000);
+  return StreamingStatus::OK;
+}
+
+StreamingStatus MockProducer::DestroyTransferChannel() {
+  std::unique_lock<std::mutex> lock(MockQueue::mutex);
+  MockQueue &mock_queue = MockQueue::GetMockQueue();
+  mock_queue.message_buffer.erase(channel_info_.channel_id);
+  mock_queue.consumed_buffer.erase(channel_info_.channel_id);
+  return StreamingStatus::OK;
+}
+
+StreamingStatus MockProducer::ProduceItemToChannel(uint8_t *data, uint32_t data_size) {
+  std::unique_lock<std::mutex> lock(MockQueue::mutex);
+  MockQueue &mock_queue = MockQueue::GetMockQueue();
+  auto &ring_buffer = mock_queue.message_buffer[channel_info_.channel_id];
+  if (ring_buffer->Full()) {
+    return StreamingStatus::OutOfMemory;
+  }
+  MockQueueItem item;
+  item.data.reset(new uint8_t[data_size]);
+  item.data_size = data_size;
+  std::memcpy(item.data.get(), data, data_size);
+  ring_buffer->Push(item);
+  return StreamingStatus::OK;
+}
+
+StreamingStatus MockProducer::RefreshChannelInfo() {
+  MockQueue &mock_queue = MockQueue::GetMockQueue();
+  channel_info_.queue_info.consumed_message_id =
+      mock_queue.queue_info_map[channel_info_.channel_id].consumed_message_id;
+  return StreamingStatus::OK;
+}
+
+
+StreamingStatus MockConsumer::ConsumeItemFromChannel(uint8_t *&data, uint32_t &data_size,
+                                         uint32_t timeout) {
+  std::unique_lock<std::mutex> lock(MockQueue::mutex);
+  MockQueue &mock_queue = MockQueue::GetMockQueue();
+  auto &channel_id = channel_info_.channel_id;
+  if (mock_queue.message_buffer.find(channel_id) == mock_queue.message_buffer.end()) {
+    return StreamingStatus::NoSuchItem;
+  }
+  if (mock_queue.message_buffer[channel_id]->Empty()) {
+    return StreamingStatus::NoSuchItem;
+  }
+  MockQueueItem item = mock_queue.message_buffer[channel_id]->Front();
+  mock_queue.message_buffer[channel_id]->Pop();
+  mock_queue.consumed_buffer[channel_id]->Push(item);
+  data = item.data.get();
+  data_size = item.data_size;
+  return StreamingStatus::OK;
+}
+
+StreamingStatus MockConsumer::NotifyChannelConsumed(uint64_t offset_id) {
+  std::unique_lock<std::mutex> lock(MockQueue::mutex);
+  MockQueue &mock_queue = MockQueue::GetMockQueue();
+  auto &channel_id = channel_info_.channel_id;
+  auto &ring_buffer = mock_queue.consumed_buffer[channel_id];
+  while (!ring_buffer->Empty() && ring_buffer->Front().seq_id <= offset_id) {
+    ring_buffer->Pop();
+  }
+  mock_queue.queue_info_map[channel_id].consumed_message_id = offset_id;
+  return StreamingStatus::OK;
+}
+
+StreamingStatus MockConsumer::RefreshChannelInfo() {
+  MockQueue &mock_queue = MockQueue::GetMockQueue();
```

Comment:
I'll add mutex here internally, please fix it in refresh channel info to avoid crash.